tracing-syslog/src/syslog.rs

```diff
@@ -0,0 +1,220 @@
+    /// System daemons without separate facility value.
+    LOG_DAEMON = libc::LOG_DAEMON,
+    /// Security/authorization messages.
+    LOG_AUTH = libc::LOG_AUTH,
+    /// Messages generated internally by `syslogd(8)`.
+    LOG_SYSLOG = libc::LOG_SYSLOG,
+    /// Line printer subsystem.
+    LOG_LPR = libc::LOG_LPR,
+    /// USENET news subsystem.
+    LOG_NEWS = libc::LOG_NEWS,
+    /// UUCP subsystem.
+    LOG_UUCP = libc::LOG_UUCP,
+    /// Clock daemon (`cron` and `at`).
+    LOG_CRON = libc::LOG_CRON,
+    /// Security/authorization messages (private).
+    LOG_AUTHPRIV = libc::LOG_AUTHPRIV,
+    /// FTP daemon.
+    LOG_FTP = libc::LOG_FTP,
+    /// Reserved for local use.
+    LOG_LOCAL0 = libc::LOG_LOCAL0,
+    /// Reserved for local use.
+    LOG_LOCAL1 = libc::LOG_LOCAL1,
+    /// Reserved for local use.
+    LOG_LOCAL2 = libc::LOG_LOCAL2,
+    /// Reserved for local use.
+    LOG_LOCAL3 = libc::LOG_LOCAL3,
+    /// Reserved for local use.
+    LOG_LOCAL4 = libc::LOG_LOCAL4,
+    /// Reserved for local use.
+    LOG_LOCAL5 = libc::LOG_LOCAL5,
+    /// Reserved for local use.
+    LOG_LOCAL6 = libc::LOG_LOCAL6,
+    /// Reserved for local use.
+    LOG_LOCAL7 = libc::LOG_LOCAL7,
+}
+
+impl Default for Facility {
+    fn default() -> Self {
+        Self::LOG_USER
+    }
+}
+
+/// `syslog` severity.
+#[allow(non_camel_case_types)]
+#[derive(Copy, Clone)]
+#[repr(i32)]
+pub enum Severity {
+    /// System is unusable.
+    LOG_EMERG = libc::LOG_EMERG,
+    /// Action must be taken immediately.
+    LOG_ALERT = libc::LOG_ALERT,
+    /// Critical conditions.
+    LOG_CRIT = libc::LOG_CRIT,
+    /// Error conditions.
+    LOG_ERR = libc::LOG_ERR,
+    /// Warning conditions.
+    LOG_WARNING = libc::LOG_WARNING,
+    /// Normal, but significant, condition.
+    LOG_NOTICE = libc::LOG_NOTICE,
+    /// Informational message.
+    LOG_INFO = libc::LOG_INFO,
+    /// Debug-level message.
+    LOG_DEBUG = libc::LOG_DEBUG,
+}
+
+impl From<Level> for Severity {
+    fn from(level: Level) -> Self {
+        match level {
+            Level::ERROR => Self::LOG_ERR,
+            Level::WARN => Self::LOG_WARNING,
+            Level::INFO => Self::LOG_NOTICE,
+            Level::DEBUG => Self::LOG_INFO,
+            Level::TRACE => Self::LOG_DEBUG,
+        }
+    }
+}
+
+/// `syslog` priority.
+#[derive(Copy, Clone, Debug)]
+struct Priority(libc::c_int);
+
+impl Priority {
+    fn new(facility: Facility, level: Level) -> Self {
+        let severity = Severity::from(level);
+        Self((facility as libc::c_int) | (severity as libc::c_int))
+    }
+}
+
+fn syslog(priority: Priority, msg: &CStr) {
+    unsafe { libc::syslog(priority.0, "%s\0".as_ptr().cast(), msg.as_ptr()) }
+}
+
+/// Creates a `syslog`-backed [`tracing_core::Collect`].
+pub fn logger(
+    identity: impl Into<Cow<'static, CStr>>,
+    options: Options,
+    facility: Facility,
+) -> impl Collect {
+    Registry::default().with(Syslog::new(identity, options, facility))
+}
```

Comment:
I'd probably omit this function and instead have the contents on line 146 in a prominent location.

Reply:
Sounds reasonable. I've added doctests showing how to create a collector on the `Syslog` type